bodo/pandas/utils.py

```diff
@@ -93,6 +98,148 @@ def cpp_table_to_df(cpp_table, arrow_schema):
     return out_df
 
 
+@functools.lru_cache
+def get_dataframe_overloads():
+    """Return a list of the functions supported on BodoDataFrame objects
+    to some degree by bodo.jit.
+    """
+    from bodo.hiframes.pd_dataframe_ext import DataFrameType
+    from bodo.numba_compat import get_method_overloads
+
+    return get_method_overloads(DataFrameType)
+
+
+@functools.lru_cache
+def get_series_overloads():
+    """Return a list of the functions supported on BodoSeries objects
+    to some degree by bodo.jit.
+    """
+    from bodo.hiframes.pd_series_ext import SeriesType
+    from bodo.numba_compat import get_method_overloads
+
+    return get_method_overloads(SeriesType)
+
+
+def get_overloads(cls_name):
+    """Use the class name of the __class__ attr of self parameter
+    to determine which of the above two functions to call to
+    get supported overloads for the current data type.
+    """
+    if cls_name == "BodoDataFrame":
+        return get_dataframe_overloads()
+    elif cls_name == "BodoSeries":
+        return get_series_overloads()
+    else:
+        assert False
+
+
+def check_args_fallback(
+    must_be_default, supported=False, package_name="pandas", fn_str=None, module_name=""
```

Comment:
Minor UX suggestion: maybe it's simpler if `supported` is a list of arguments instead of a flag and only one of `must_be_default` and `supported` can be set.